js_modules/dagit/packages/core/src/launchpad/LaunchpadSession.tsx

```diff
@@ -495,6 +505,29 @@ const LaunchpadSessionContainer: React.FC<LaunchpadSessionContainerProps> = (pro
     return null;
   }, [currentSession]);
 
+  const LaunchButtonContainer: React.FC = ({children}) => {
```

Comment:
Defining a component inside another component is a bit of a React code smell. React won't realize that two renders are producing the same output because the LaunchButtonContainer is an instance of a separate component every render. It'll unmount + remount this subtree, even if there haven't been any prop changes.

I think it'd be ideal to move this outside of this component, and make launchpadType a prop of it. (So you'd say `<LaunchButtonContainer launchpadType="asset" >` down below)

Reply:
done